File: client/src/app/shared/models/product.ts
export interface IProduct {
  id: number;
  name: string;
  description: string;
  price: number;
  sifra: string;
  oldPrice: number;
  pictureUrl: string;
  productType: string;
  productBrand: string;
  stock: number;
  izdvojen: boolean;
  novo: boolean;
  sale: boolean;
  photos: IPhoto[];
}

export interface IPhoto {
  id: number;
  pictureUrl: string;
  fileName: string;
  isMain: boolean;
}

export interface IProductToCreate {
  name: string;
  description: string;
  price: number;
  oldPrice: number;
  pictureUrl: string;
  productTypeId: number;
  productBrandId: number;
  stock: number;
}

export class ProductFormValues implements IProductToCreate {
  name = '';
  description = '';
  price = 0;
  sifra = '';
  oldPrice = 0;
  pictureUrl = '';
  productBrandId: number;
  productTypeId: number;
  stock = 0;

  constructor(init?: ProductFormValues) {
    Object.assign(this, init);
  }
}

File: Core/Entities/Product.cs
using System.Collections.Generic;
using System.Linq;

namespace Core.Entities
{
    public class Product : BaseEntity
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public decimal OldPrice { get; set; }
        public bool? Izdvojen { get; set; } = null;
        public string Sifra { get; set; } = "";
        public bool? Novo { get; set; } = null;
        public bool? Sale { get; set; } = null; 
        public int Stock { get; set; } = 0;
        public ProductType ProductType { get; set; }
        public int ProductTypeId { get; set; }
        public ProductBrand ProductBrand { get; set; }
        public int ProductBrandId { get; set; }
        private readonly List<Photo> _photos = new List<Photo>();
        public List<Photo> Photos => _photos.ToList();

        public void AddPhoto(string pictureUrl, string fileName, bool isMain = false)
        {
            var photo = new Photo
            {
                FileName = fileName,
                PictureUrl = pictureUrl
            };
            
            if (_photos.Count == 0) photo.IsMain = true;
            
            _photos.Add(photo);
        }

        public void RemovePhoto(int id)
        {
            var photo = _photos.Find(x => x.Id == id);
            _photos.Remove(photo);
        }

        public void SetMainPhoto(int id)
        {
            var currentMain = _photos.SingleOrDefault(item => item.IsMain);
            foreach (var item in _photos.Where(item => item.IsMain))
            {
                item.IsMain = false;
            }
            
            var photo = _photos.Find(x => x.Id == id);
            if (photo != null)
            {
                photo.IsMain = true;
                if (currentMain != null) currentMain.IsMain = false;
            }
        }
    }
}